### DISCUSSION TITLE
Question about dependency between build-stage and stripping

### DISCUSSION THREAD

**Discussion by zazaho** [rpm-software-management/rpm#2805] [Q&A]
Hello,

I have a question. I have packaged a software where I have not used a build stage because there is a bug in its cmake logic that makes that the compilation repeats during install. I noticed that the packaged binaries are not stripped. Is this according to specifications?

Basically the relevant part of my spec file looks like this:
```
%setup -q
install rpmbuilddir

%install
pushd rpmbuilddir
%cmake ..
%make_install
popd

%files
```

Would the proper work around be to include an empty `%build`  stage ?

Thank you for your help

Sacha

**Comment by dmnks** [comment]
The absence of the `%build` script shouldn't have any effect on whether the stripping occurs or not. In fact, the `brp-strip` script is called *after* the `%install` script.

I'd suspect there to be some other reason the binaries aren't stripped for you. Can you share the `rpmbuild` output?

**Reply by zazaho** [reply]
here an example output:
[rpmbuild_expat_log.txt](https://github.com/rpm-software-management/rpm/files/14184807/rpmbuild_expat_log.txt)

the result from this spec file:
[expat.spec.txt](https://github.com/rpm-software-management/rpm/files/14184817/expat.spec.txt)

The binaries are not stripped.

Thank you for your help

**Reply by zazaho** [reply]
if I include a empty build stage the binaries are stripped 
[rpmbuild_expat__with_build_stage_log.txt](https://github.com/rpm-software-management/rpm/files/14184852/rpmbuild_expat__with_build_stage_log.txt)

[expat_with_build.spec.txt](https://github.com/rpm-software-management/rpm/files/14184877/expat_with_build.spec.txt)

**Reply by zazaho** [reply]
![image](https://github.com/rpm-software-management/rpm/assets/16631687/7b62c319-4f8d-4e07-8e34-4d4247e81fe2)
